Ghidra/Framework/Utility/src/main/java/ghidra/framework/ColorProperties.java

```diff
@@ -0,0 +1,106 @@
+     */
+    public static final String COLOR_PROPERTY_NAME = "Color.properties";
+    public static final String COLOR_PROPERTY_FILE = "/RuntimeScripts/Common/support/Color.properties";
+    public static final String COLOR_PROPERTY_FILE_SUB = "/support/Color.properties";
+
+    /**
+     * Creates a new color properties from the given config properties file.
+     *
+     * @param colorPropertiesFile The color properties file.
+     * @throws IOException If there was a problem loading/reading a discovered properties file.
+     */
+    public ColorProperties(ResourceFile colorPropertiesFile) throws IOException {
+
+        if (!colorPropertiesFile.exists()) {
+            throw new FileNotFoundException(
+                    COLOR_PROPERTY_NAME + " file does not exist: " + colorPropertiesFile);
+        }
+        try (InputStream in = colorPropertiesFile.getInputStream()) {
+            load(in);
+        }
+    }
+
+    /**
+     * Creates a new color properties from the color properties files found
+     * in the given application root directories.  If multiple color properties files
+     * are found, the properties from the files will be combined.  If duplicate keys exist,
+     * the newest key encountered will overwrite the existing key.
+     *
+     * @param applicationRootDirs The application root directories to look for the properties files in.
+     * @throws IOException If there was a problem loading/reading a discovered properties file.
+     */
+    public ColorProperties(Collection<ResourceFile> applicationRootDirs) throws IOException {
+        boolean found = false;
+        // Application installation directory
+        ResourceFile applicationInstallationDir = applicationRootDirs.iterator().next().getParentFile();
+        if (SystemUtilities.isInDevelopmentMode()) {
+            for (ResourceFile appRoot : applicationRootDirs) {
+                ResourceFile colorPropertiesFile = new ResourceFile(appRoot, COLOR_PROPERTY_FILE);
+                if (colorPropertiesFile.exists()) {
+                    try (InputStream in = colorPropertiesFile.getInputStream()) {
+                        load(in);
+                        found = true;
+                    }
+                }
+            }
+        }
+        else {
+            ResourceFile colorPropertiesFile = new ResourceFile(applicationInstallationDir, COLOR_PROPERTY_FILE_SUB);
+            if (colorPropertiesFile.exists()) {
+                try (InputStream in = colorPropertiesFile.getInputStream()) {
+                    load(in);
+                    found = true;
+                }
+            }
+        }
+        if (!found) {
+            throw new IOException(COLOR_PROPERTY_NAME + " was not found!");
+        }
+    }
+
+    /**
+     * Get Properties from Color.properties by key
+     *
+     * @param key Color.properties key
+     * @return Color Object
+     * */
+	public Color readColorFromProperties(String key) {
```

Comment:
I don't see any reason why we need such a long name. A simple `readColor` or `findColor` should do.